### PR TITLE
Remake watchOS app

### PR DESCRIPTION
**Pull request by whoiswillma**
## Overview

Reimplement the Eatery watchOS app using SwiftUI!

Screen Shot 2020-01-19 at 10 34 14 Screen Shot 2020-01-19 at 10 34 29 Screen Shot 2020-01-19 at 10 34 41 

## Test Coverage

I tested the app on the simulator and it works fine. I minimally tested it on a physical device (since I had to borrow an Apple Watch). 

There may still be issues with location permissions. 

## Next Steps

There are a few missing features:
- Preserve filter and sort settings between launches
- Add a loading indicator when Eateries are loading 

To fix:
- If the Eatery doesn't have a menu title (e.g. Bear Necessities, Louie's), then the menu title displays as "General", "Menu", or ""
- And there are unknown bugs to be ironed out.

### CODE REVIEW

File: Eatery Watch App Extension/Notification (Unused)/NotificationController.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    override func didDeactivate() {
+        // This method is called when watch view controller is no longer visible
+        super.didDeactivate()
+    }
+
+    override func didReceive(_ notification: UNNotification) {
+        // This method is called when a notification needs to be presented.
+        // Implement it if you use a dynamic notification interface.
+        // Populate your dynamic notification interface as quickly as possible.
+    }
```
Comment (kevinchan159) [inline Eatery Watch App Extension/Notification (Unused)/NotificationController.swift]: are these necessary? you dont seem to add any additional logic to them

File: Eatery Watch App Extension/UI/Eateries/CampusEateriesView.swift [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+//  Copyright © 2020 CUAppDev. All rights reserved.
+//
+
+import Combine
+import CoreLocation
+import SwiftUI
+
+private class CampusEateriesData: ObservableObject {
+
+    @Published var campusEateries: [CampusEatery] = []
```
Comment (kevinchan159) [inline Eatery Watch App Extension/UI/Eateries/CampusEateriesView.swift:15]: i haven't dug into SwiftUI yet but just looked up what this is and this is really cool!

File: Eatery Watch App Extension/UI/Eateries/CampusEateriesView.swift [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+    @State private var sortMethod: SortMethod = .name
+    @State private var areaFilter: Area?
+
+    @State private var errorInfo: ErrorInfo?
+
+    @State private var firstAppearance = true
+
+    var body: some View {
+        let open = sort(filter(self.viewData.openEateries))
```
Comment (kevinchan159) [inline Eatery Watch App Extension/UI/Eateries/CampusEateriesView.swift]: i think `open` is already a keyword? might be better to rename this variable

File: Eatery Watch App Extension/UI/Eateries/CampusEateriesView.swift [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        if let areaFilter = self.areaFilter {
+            return eateries.filter { eatery in
+                eatery.area == areaFilter
+            }
+        } else {
+            return eateries
+        }
+    }
+
+    private func sort(_ eateries: [CampusEatery]) -> [CampusEatery] {
```
Comment (kevinchan159) [inline Eatery Watch App Extension/UI/Eateries/CampusEateriesView.swift]: can we rename this to `sorted`. i believe `sort` is typically implied to sort the array in place without returning any value

File: Eatery Watch App Extension/UI/Eatery/CampusEateryMealsView.swift [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+//  Eatery Watch App Extension
+//
+//  Created by William Ma on 1/7/20.
+//  Copyright © 2020 CUAppDev. All rights reserved.
+//
+
+import SwiftUI
+
+/// A list of menus that
+struct CampusEateryMealsView: View {
```
Comment (kevinchan159) [inline Eatery Watch App Extension/UI/Eatery/CampusEateryMealsView.swift:12]: nit: add new line at the beginning and closing of `struct`
same for the other structs in the PR

File: Eatery/AppDelegate.swift [hunk 6]
```diff
@@ -45,7 +45,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         #else
             print("RUNNING EATERY IN RELEASE CONFIGURATION")
         #endif
-
+
```
Comment (kevinchan159) [inline Eatery/AppDelegate.swift]: remove whitespace

**Review by kevinchan159 — APPROVED** [review]
wow this is a huge PR
this is AMAZING. thanks so much for working on this 😍. I almost forgot we even had a watchOS app LOL

File: Eatery Watch App Extension/Notification (Unused)/NotificationController.swift (hunk 1 above)
Comment (whoiswillma) [reply Eatery Watch App Extension/Notification (Unused)/NotificationController.swift]: I'm not sure what these are used for, but they were added to the app and I was scared to remove them. I've taken them out and tested the app lightly again.

**Review by ReadeP — APPROVED** [review]
This is awesome, great work! 🤩

**Pull request merged**